tests/api2/test_port_delegates.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+
+from middlewared.service_exception import ValidationErrors
+from middlewared.test.integration.utils import call
+
+
+PAYLOAD = (
+    ('s3.config', 's3.update', ['bindport', 'console_bindport'], {'access_key': '12345678', 'secret_key': '123456789'}),
+    ('ftp.config', 'ftp.update', ['port'], {}),
+    ('webdav.config', 'webdav.update', ['tcpport', 'tcpportssl'], {}),
+    ('rsyncd.config', 'rsyncd.update', ['port'], {}),
+)
+
+
+def test_port_delegate_validation_with_invalid_ports():
+    in_use_ports = []
+    for entry in call('port.get_in_use'):
+        in_use_ports.extend(filter(lambda i: i > 1024, entry['ports']))
+
+    assert in_use_ports != [], 'No in use ports retrieved'
+
+    for config_method, method, keys, payload in PAYLOAD:
```

Comment:
It's better to use `pytest.mark.parametrize` and try each payload in a separate test. If you use `for` loop, one payload failure will result to subsequent not being tested.